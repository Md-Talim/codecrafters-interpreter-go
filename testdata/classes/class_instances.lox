// Test Case 1
// Creating multiple instances of the same class
class Robot {}
var r1 = Robot();
var r2 = Robot();

print "Created multiple robots:";
print r1;
print r2;

// Test Case 2
class Wizard {}
class Dragon {}

// Instantiating classes in a function should work
fun createCharacters() {
  var merlin = Wizard();
  var smaug = Dragon();
  print "Characters created in fantasy world:";
  print merlin;
  print smaug;
  return merlin;
}

var mainCharacter = createCharacters();
// An instance of a class should be truthy
if (mainCharacter) {
  print "The main character is:";
  print mainCharacter;
} else {
  print "Failed to create a main character.";
}

// Test Case 3
class Superhero {}

var count = 0;
while (count < 3) {
  var hero = Superhero();
  print "Hero created:";
  print hero;
  count = count + 1;
}

print "All heroes created!";
