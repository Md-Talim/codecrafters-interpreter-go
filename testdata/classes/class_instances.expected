Created multiple robots:
Robot instance
Robot instance
Characters created in fantasy world:
Wizard instance
Dragon instance
The main character is:
Wizard instance
Hero created:
Superhero instance
Hero created:
Superhero instance
Hero created:
Superhero instance
All heroes created!